### PR TITLE
Improved xyzpub screen with some address details

### PR DESCRIPTION
**Pull request by Thierry61**
- global balance in header
- balance and transaction count in each address
- possibility to use extended public keys in btcFun.js

### CODE REVIEW

**Comment by janoside** [comment]
Excited about this! I'll be checking it out shortly.

**Comment by pointbiz** [comment]
I'm also very excited about this!

**Comment by pointbiz** [comment]
I'm also very excited about this!

**Comment by janoside** [comment]
Nice work @Thierry61! Merged and tweaked. Looking forward to continuing to build on this.

File: views/extended-public-key.pug [hunk 1]
```diff
@@ -24,11 +24,10 @@ block content
 			+summaryItem("BIP32 Path", "BIP32 defines a standard for deriving child keys from an extended parent key. This 'path', also known as the 'derivation path', comes from that standard and instructs software how to generate child keys.")
 				| #{bip32Path}
 
-			if (balance)
-				+summaryItem("Balance")
-
-			if (changeBalance)
-				+summaryItem("Change Balance")
+			+summaryItem("Balance")
+				span(class=(balanceSat > 0 ? "text-success" : false))
```
Comment (pointbiz) [inline views/extended-public-key.pug:28]: The challenge with the balance appearing here is that it's the balance of the first 20 receive and 20 change addresses. A business wallet (or power user) could have more addresses. Somehow the user would need to know the limitations of this balance calculation.

**Review by pointbiz — COMMENTED** [review]
This UI is challenging. My opinion is this PR would be ready by removing the main balance from the display. And adding the loop on the transaction count instead of the hard coded 100 transactions. This way the user doesn't see an inaccurate main balance and the user gets new functionality by seeing the tx count and balance for each address. 

Your code was easy to read.

**Comment by pointbiz** [comment]
> Nice work @Thierry61! Merged and tweaked. Looking forward to continuing to build on this.

What's your thoughts on an additional pug view that is linked from the xyzpub page which would show all the transactions using pagination (very similar to the address view pug)? Maybe the main balance would be calculated and show there?

**Comment by janoside** [comment]
@pointbiz I might agree. I'm trying to think through UX issues in this area, but I definitely agree it's not "done" by any means. I'm just happy to have something to build on top of.

**Comment by pointbiz** [comment]
I really like the per address tx count and balance. 
Kudos to @Thierry61

File: views/extended-public-key.pug (hunk 1 above)
Comment (Thierry61) [reply views/extended-public-key.pug:28]: Note that @janoside has subsequently clarified balance meaning by displaying current page address range  (cf. commit 9fdc075)

**Pull request merged**